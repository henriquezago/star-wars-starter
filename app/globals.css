:root {
  --foreground: #ededed;

  /* Colors */
  --gainsboro: #dadada;
  --darkgray: rgba(132, 132, 132, 0.75);
  --gray-22: #383838;
  --white: #fff;
  --black: #000;
  --silver: #c4c4c4;
  --green: #0ab463;
  --blue: #0094ff;
}

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

html,
body {
  max-width: 100vw;
  overflow-x: hidden;
  font-family: Montserrat, sans-serif;
}

.header {
  box-shadow: 0 2px 0 0 var(--gainsboro);
  background-color: #fff;
  padding: 0.875rem 0;
  text-align: center;
  font-size: 1.125rem;
  font-weight: bold;
  color: #0ab463;
}

body {
  color: var(--black);
  background: var(--foreground);
}

a {
  color: var(--blue);
  text-decoration: none;
  font-size: 0.875rem;
  font-weight: normal;
}
