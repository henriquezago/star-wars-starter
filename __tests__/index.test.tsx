import { expect } from "@jest/globals";
import { fireEvent, render, screen } from "@testing-library/react";

import Home from "../app/page";

describe("Home", () => {
  beforeEach(() => {
    global.fetch = jest.fn().mockResolvedValue({
      json: async () => {
        return {
          results: [],
        };
      },
    });
  });

  it("renders a heading", () => {
    const { container } = render(<Home />);

    expect(container).toMatchSnapshot();
  });

  it("should call fetch on people's api on form submit", () => {
    render(<Home />);

    const input = screen.getByTestId("search-input");
    fireEvent.change(input, { target: { value: "Luke" } });

    const submitButton = screen.getByText("Search");
    fireEvent.click(submitButton);

    expect(fetch).toHaveBeenCalledWith("/api/people?search=Luke");
  });

  it("should call fetch on movies's api on form submit", () => {
    render(<Home />);

    const moviesRadio = screen.getByLabelText("Movies");
    fireEvent.click(moviesRadio);
    const input = screen.getByTestId("search-input");
    fireEvent.change(input, { target: { value: "Luke" } });

    const submitButton = screen.getByText("Search");
    fireEvent.click(submitButton);

    expect(fetch).toHaveBeenCalledWith("/api/movies?search=Luke");
  });
});
