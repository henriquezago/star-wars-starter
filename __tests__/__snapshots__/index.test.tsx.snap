// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a heading 1`] = `
<div>
  <main
    class="main"
  >
    <div
      class="card filtersCard"
    >
      <h1
        class="title"
      >
        What are you searching for?
      </h1>
      <div
        class="inputsContainer"
      >
        <input
          checked=""
          id="people"
          name="searchType"
          type="radio"
          value="people"
        />
        <label
          for="people"
        >
          People
        </label>
        <input
          id="movies"
          name="searchType"
          type="radio"
          value="movies"
        />
        <label
          for="movies"
        >
          Movies
        </label>
      </div>
      <form>
        <input
          class="textInput fullWidth"
          data-testid="search-input"
          placeholder="e.g. Chewbacca, Yoda, Boba Fett"
          type="text"
          value=""
        />
        <button
          class="button fullWidth"
          disabled=""
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="card card"
    >
      <h2
        class="title"
      >
        Results
      </h2>
      <div
        class="horizontalDivider"
      />
      <div
        class="emptyState"
      >
        <h3>
          Searching...
        </h3>
      </div>
    </div>
  </main>
</div>
`;
